homeassistant/components/zha/core/channels/base.py

```diff
@@ -96,7 +104,7 @@ class ChannelStatus(Enum):
 class ZigbeeChannel(LogMixin):
     """Base channel for a Zigbee cluster."""
 
-    REPORT_CONFIG: tuple[dict[int | str, tuple[int, int, int | float]]] = ()
+    REPORT_CONFIG: Sequence[ReportConfig] = ()
```

Comment:
The current type hint for REPORT_CONFIG isn't valid and causes all sort of issues:
- some classes use a list and some classes use a tuple so I decided to go for Sequence
- I couldn't find any instance where the key was an integer, it seems to be always "attr" and "config"
- the dict values for `attr` are strings so that should have been an option in the dict values

Reply:
> I also tried Sequence[dict[str, str | tuple[int, int, int]]] but that caused issues due to the use of constants in the various REPORT_CONFIG_XXX tuples

Can you provide lines/examples which are causing the issue?


I'd think it should be `Sequence[dict[st | int,  tuple[int, int, int | float]]]`


> some classes use a list and some classes use a tuple so I decided to go for Sequence

I think we should change lists to tuples

Reply:
Do you not like the `TypedDict`? I think that the cleanest way to do it right now.

Reply:
Not what I meant :) I'm totally fine with the TypeDict. 
Just trying to workaround the issue with tuple constants. And beyond basic typing hints, it is still a bit of magic to me. 

I'm just asking if the TypeDict stays in place, but
`config: Tuple[int, int, int]` becomes `config: ReportConfig`, would MyPy still complain on `AttrReportConfig(attr = "measured_value", config = CONFIG_REPORT_OP)` ?

Assuming COFIG_REPORT_OP was defined as per earlier message?

Reply:
MyPy doesn't complain with the `TypedDict`, it only complains with the standard `dict`.